File: client/lib/service/product_service.dart
import 'package:app_food_scan/data/product_dao.dart';

import '../mvc/models/product_model.dart';

class ProductService {
  final ProductDAO productDAO;

  ProductService(this.productDAO);

  Future<List<ProductModel>> index() async {
    final List<Map<String, dynamic>> maps = await productDAO.read();
    return List.generate(maps.length, (index) => ProductModel(idProduct: maps[index][ProductModel.idProductColumn]));
  }

  Future<ProductModel> create(ProductModel product) async {
    await productDAO.create(product);
    return product;
  }
}

File: server/src/main/java/com/example/app_food_scan/services/ProductService.java
package com.example.app_food_scan.services;

import com.example.app_food_scan.models.Product;
import com.example.app_food_scan.services.data.AllergenTotals;

import java.util.List;

public interface ProductService {

    void reportAllergens(String idDevice, Product product);

    List<AllergenTotals> getTotalsByAllergen(String idProduct);

    Product getProduct(String idProduct);

}
